cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/sync/AdminRequest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet.handler.sync;
+
+import com.linkedin.kafka.cruisecontrol.KafkaCruiseControl;
+import com.linkedin.kafka.cruisecontrol.detector.notifier.AnomalyType;
+import com.linkedin.kafka.cruisecontrol.servlet.KafkaCruiseControlServlet;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.AdminParameters;
+import com.linkedin.kafka.cruisecontrol.servlet.response.AdminResult;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class AdminRequest extends AbstractSyncRequest {
+  private static final Logger LOG = LoggerFactory.getLogger(AdminRequest.class);
+  private final KafkaCruiseControl _kafkaCruiseControl;
+  private final AdminParameters _parameters;
+
+  public AdminRequest(KafkaCruiseControlServlet servlet, AdminParameters parameters) {
+    super(servlet);
+    _kafkaCruiseControl = servlet.asyncKafkaCruiseControl();
+    _parameters = parameters;
+  }
+
+  /**
+   * Handle the admin requests:
+   * <ul>
+   * <li>Dynamically change the partition and leadership concurrency of an ongoing execution. Has no effect if Executor
+   * is in {@link com.linkedin.kafka.cruisecontrol.executor.ExecutorState.State#NO_TASK_IN_PROGRESS} state.</li>
+   * <li>Enable/disable the specified anomaly detectors.</li>
+   * <li>Drop selected recently removed/demoted brokers.</li>
+   * </ul>
+   *
+   * @return Admin response.
+   */
+  @Override
+  protected AdminResult handle() {
+    String ongoingConcurrencyChangeRequest = "";
+    // 1.1. Change inter-broker partition concurrency.
+    Integer concurrentInterBrokerPartitionMovements = _parameters.concurrentInterBrokerPartitionMovements();
+    if (concurrentInterBrokerPartitionMovements != null) {
+      _kafkaCruiseControl.setRequestedInterBrokerPartitionMovementConcurrency(concurrentInterBrokerPartitionMovements);
+      ongoingConcurrencyChangeRequest += String.format("Inter-broker partition movement concurrency is set to %d%n",
+                                                       concurrentInterBrokerPartitionMovements);
+      LOG.warn("Inter-broker partition movement concurrency is set to: {} by user.", concurrentInterBrokerPartitionMovements);
+    }
+    // 1.2. Change leadership concurrency.
+    Integer concurrentLeaderMovements = _parameters.concurrentLeaderMovements();
+    if (concurrentLeaderMovements != null) {
+      _kafkaCruiseControl.setRequestedLeadershipMovementConcurrency(concurrentLeaderMovements);
+      ongoingConcurrencyChangeRequest += String.format("Leadership movement concurrency is set to %d%n", concurrentLeaderMovements);
+      LOG.warn("Leadership movement concurrency is set to: {} by user.", concurrentLeaderMovements);
```

Comment:
`warn` -> `info`?

Reply:
All admin request logs were intentionally set to `warn` level as bad `admin` requests have the power to overload / kill / harm the Kafka cluster.

-- That being said, I don't have a strong opinion on this. Do you think we should change all / some logs in `Admin` request?

Reply:
My experience in debugging is that when I look for suspicious things in log, I always grep for `WARN`/`ERROR`/`Exception` first. I feel `WARN` still means things are bad(to some extent, but not as bad as `Error`...).

Maybe we can keep `None of the user-requested brokers to drop ({}) are in the recently demoted broker set ({}).` as `warn` and change other to `info`?

Reply:
Sure, that makes sense -- I will make the changes.